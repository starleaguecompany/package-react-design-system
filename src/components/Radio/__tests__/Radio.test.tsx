import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Radio } from '../../../index'

describe('Radio', () => {
  test('should render correctly with no props', () => {
    render(<Radio />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByTestId('Radio')).toHaveClass('container', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<Radio id="test-id" className="test-class" />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByTestId('Radio')).toHaveClass('test-class')
    expect(screen.getByRole('radio')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with label', () => {
    render(<Radio>Radio label</Radio>)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByRole('radio')).toBeVisible()
    expect(screen.getByText('Radio label')).toBeVisible()
    expect(screen.getByText('Radio label')).toHaveClass('label')
  })

  test('should render correctly with name prop', () => {
    render(<Radio name="Radio name" />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByRole('radio')).toHaveAttribute('name', 'Radio name')
  })

  test('should render correctly with value prop', () => {
    render(<Radio value="Radio value" />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByRole('radio')).toHaveAttribute('value', 'Radio value')
  })

  test('should render correctly with checked prop', () => {
    render(<Radio checked />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByTestId('Radio')).toHaveClass('checked')
    expect(screen.getByRole('radio')).toBeChecked()
  })

  test('should render correctly with disabled prop', () => {
    render(<Radio disabled />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByTestId('Radio')).toHaveClass('disabled')
    expect(screen.getByRole('radio')).toBeDisabled()
  })

  test('should render correctly with readOnly prop', () => {
    render(<Radio readOnly />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    expect(screen.getByTestId('Radio')).toHaveClass('readOnly')
  })

  test('should execute onChangeCallback on click radio', () => {
    const onChangeCallback = jest.fn()
    render(<Radio onChange={onChangeCallback} />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    userEvent.click(screen.getByRole('radio'))
    expect(onChangeCallback).toBeCalledTimes(1)
  })

  test('should render correctly on external value update', () => {
    const onChangeCallback = jest.fn()
    const { rerender } = render(<Radio checked={false} onChange={onChangeCallback} />)

    expect(screen.getByTestId('Radio')).toMatchSnapshot()
    rerender(<Radio checked={true} onChange={onChangeCallback} />)
    expect(screen.getByRole('radio')).toBeChecked()
    expect(onChangeCallback).toBeCalledTimes(0)
  })
})
