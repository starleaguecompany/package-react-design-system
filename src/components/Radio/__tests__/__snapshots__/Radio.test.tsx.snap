// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should execute onChangeCallback on click radio 1`] = `
<label
  class="container"
  data-qa="Radio"
>
  <input
    class="input"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Radio should render correctly on external value update 1`] = `
<label
  class="container"
  data-qa="Radio"
>
  <input
    class="input"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Radio should render correctly with attributes 1`] = `
<label
  class="test-class container"
  data-qa="Radio"
>
  <input
    class="input"
    id="test-id"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Radio should render correctly with checked prop 1`] = `
<label
  class="container checked"
  data-qa="Radio"
>
  <input
    checked=""
    class="input"
    type="radio"
    value=""
  />
  <div
    class="control checked"
  >
    <svg
      class="icon"
      data-qa="Check"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6 11 3 3 6-7"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</label>
`;

exports[`Radio should render correctly with disabled prop 1`] = `
<label
  class="container disabled"
  data-qa="Radio"
>
  <input
    class="input"
    disabled=""
    type="radio"
    value=""
  />
  <div
    class="control disabled"
  />
</label>
`;

exports[`Radio should render correctly with label 1`] = `
<label
  class="container"
  data-qa="Radio"
>
  <input
    class="input"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
  <span
    class="label"
  >
    Radio label
  </span>
</label>
`;

exports[`Radio should render correctly with name prop 1`] = `
<label
  class="container"
  data-qa="Radio"
>
  <input
    class="input"
    name="Radio name"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Radio should render correctly with no props 1`] = `
<label
  class="container"
  data-qa="Radio"
>
  <input
    class="input"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Radio should render correctly with readOnly prop 1`] = `
<label
  class="container readOnly"
  data-qa="Radio"
>
  <input
    class="input"
    type="radio"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Radio should render correctly with value prop 1`] = `
<label
  class="container"
  data-qa="Radio"
>
  <input
    class="input"
    type="radio"
    value="Radio value"
  />
  <div
    class="control"
  />
</label>
`;
